index.js

```diff
@@ -136,36 +200,47 @@ function nowServing(line) {
 }
 
 const line = []
+let ticketNumber = 0 
+function takeATicketNumber(line){
+  //let ticketNumber = 0  will start 0 every time
+   ticketNumber ++
+   line.push(ticketNumber)
+   return `Welcome. You are ticket number ${ticketNumber}.`
 
-// ***** Scope & Closures - Question 1 *****
+}
 
+// ***** Scope & Closures - Question 1 *****
 
+git
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// // => `Welcom e. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
+const newLine = []
 // const takeATicketNumberFunc = ticketNumberGeneratorFunc()
+// ticketNumberGeneratorFunc(){
```

Comment:
Take a look at the solution, but remember that in JS, functions are objects!
Which means it can be returned from a function invocation!